app/models/concerns/with_optimistic_retry.rb

```diff
@@ -0,0 +1,26 @@
+# http://apidock.com/rails/ActiveRecord/Persistence/reload
+module WithOptimisticRetry
+
+  private
+
+  def with_optimistic_retry(retries=nil)
+    retries ||= 10
+    begin
+      yield
+    rescue ActiveRecord::StaleObjectError => e
+      begin
+        # Reload lock_version in particular.
+        reload
+        retries -= 1
+      rescue ActiveRecord::RecordNotFound
+        # If the record is gone there is nothing to do.
```

Comment:
Sorry I mean should it re-raise the error after catching it. It shouldn't happen since don't ever delete queues at the moment right?